caddyhttp/httpserver/server.go

```diff
@@ -507,16 +507,44 @@ func (s *Server) Stop() error {
 // OnStartupComplete lists the sites served by this server
 // and any relevant information, assuming caddy.Quiet == false.
 func (s *Server) OnStartupComplete() {
+
 	if caddy.Quiet {
 		return
 	}
+
+	s.OutputSiteInfo()
+
+}
+
+func (s *Server) OutputSiteInfo() {
+
+	sitesByPort := make(map[string][]SiteConfig)
 	for _, site := range s.sites {
-		output := site.Addr.String()
-		if caddy.IsLoopback(s.Address()) && !caddy.IsLoopback(site.Addr.Host) {
-			output += " (only accessible on this machine)"
+		// Group Sites together by port before outputing
+		sitesArray := sitesByPort["port "+site.Port()]
```

Comment:
Any reason why "port" needs to be part of the map key? Since all keys are ports anyway...